src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,57 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static int GROWTH_FACTOR;
+    private T[] stack;
+    private int size;
+    private int capacity;
 
+    public MateStack() {
+        GROWTH_FACTOR = 2;
+        capacity = 10;
+        stack = (T[]) new Object[capacity];
+        size = 0;
+    }
+
+    public MateStack(int defaultCapacity, int growthFactor) {
+        GROWTH_FACTOR = growthFactor;
+        capacity = defaultCapacity;
+        stack = (T[]) new Object[defaultCapacity];
+        size = 0;
+    }
+
+    public void push(T value) {
+        if (size >= capacity) {
+            grow();
+        }
+        stack[size] = value;
+        size++;
```

Comment:
![image](https://github.com/mate-academy/jv-stack/assets/140905715/0c70eb44-360b-427d-abf2-148ad920a654)

```suggestion
        stack[size++] = value;
```